src/librustc/hir/mod.rs

```diff
@@ -539,9 +538,10 @@ pub enum PatKind {
     /// The `bool` is `true` in the presence of a `..`.
     Struct(Path, HirVec<Spanned<FieldPat>>, bool),
 
-    /// A tuple struct/variant pattern `Variant(x, y, z)`.
-    /// "None" means a `Variant(..)` pattern where we don't bind the fields to names.
-    TupleStruct(Path, Option<HirVec<P<Pat>>>),
+    /// A tuple struct/variant pattern `Variant(x, y, .., z)`.
+    /// If the `..` pattern fragment presents, then `Option<usize>` denotes its position.
```

Comment:
One allocation, the variant itself is more compact (6 ptrs vs 4 ptrs), simpler traversal (no `vec1.chain(vec2)`).